File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/DefaultSendTelemetry.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class DefaultSendTelemetry
    {
        public string? TelemetryName { get; set; }

        public string? TelemetryValue { get; set; }

        public TestCaseDuration? Timeout { get; set; }

        public int? Qos { get; set; }
    }
}

File: rust/azure_iot_operations_protocol/tests/metl/default_send_telemetry.rs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use serde::Deserialize;

use crate::metl::test_case_duration::TestCaseDuration;

#[derive(Deserialize, Debug)]
pub struct DefaultSendTelemetry {
    #[serde(rename = "telemetry-name")]
    pub telemetry_name: Option<String>,

    #[serde(rename = "telemetry-value")]
    pub telemetry_value: Option<String>,

    #[serde(rename = "timeout")]
    pub timeout: Option<TestCaseDuration>,

    #[serde(rename = "qos")]
    pub qos: Option<i32>,
}
